client/page_util.go

```diff
@@ -25,13 +26,13 @@ func ReadLimits(pageSize *int, limit *int) int {
 	}
 }
 
-func GetNext(baseUrl string, response interface{}, getNextPage func(nextPageUri string) (interface{}, error)) (interface{}, error) {
+func GetNextWithContext(ctx context.Context, baseUrl string, response interface{}, getNextPage func(ctx context.Context, nextPageUri string) (interface{}, error)) (interface{}, error) {
```

Comment:
I think we need to add this as a second method, similar to the SendRequestWithContext, otherwise we're going to break everyone's clients.